apisix/discovery/consul_kv.lua

```diff
@@ -0,0 +1,416 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local require            = require
+local local_conf         = require("apisix.core.config_local").local_conf()
+local core               = require("apisix.core")
+local resty_consul       = require('resty.consul')
+local cjson              = require('cjson')
+local http               = require('resty.http')
+local ipairs             = ipairs
+local error              = error
+local ngx                = ngx
+local unpack             = unpack
+local string_gmatch      = string.gmatch
+local tonumber           = tonumber
+local pairs              = pairs
+local ipairs             = ipairs
+local ngx_timer_at       = ngx.timer.at
+local log                = core.log
+local ngx_decode_base64  = ngx.decode_base64
+
+local applications = core.table.new(0, 5)
+local default_service
+local default_weight
+local default_prefix_rule
+local skip_keys_map = core.table.new(0, 1)
+
+local events
+local events_list
+local consul_apps
+
+local schema = {
+    type = "object",
+    properties = {
+        servers = {
+            type = "array",
+            minItems = 1,
+            items = {
+                type = "string",
+            }
+        },
+        delay = {type = "integer", minimum = 1, default = 3},
+        connect_type = {
```

Comment:
Why not using `keepalive`, which is more semantic.
```suggestion
        keepalive = {
            type = "boolean",
            default = true,
        }
```

Reply:
Good idea!